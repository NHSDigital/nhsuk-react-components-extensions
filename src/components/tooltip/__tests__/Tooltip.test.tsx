import React from 'react';
// import { mount } from 'enzyme';
import {render} from '@testing-library/react'
import Tooltip from '..';

describe('Tooltip', () => {
  it('Matches Snapshot', () => {
    const component = render(<Tooltip tooltip="Tooltip!">Hover Over Me</Tooltip>);
    expect(component).toMatchSnapshot();
  });

  it('Passes through IDs', () => {
    const componentWithoutId = render(<Tooltip tooltip="Tooltip!">Hover Over Me</Tooltip>);
    const componentWithId = render(
      <Tooltip tooltip="Tooltip!" id="test-id">
        Hover Over Me
      </Tooltip>,
    );

    // 3363 - note - do these need to be more complex???
    expect(componentWithoutId.container.querySelector('#test-id')).toBeNull();
    expect(componentWithoutId.container).toMatchSnapshot();

    expect(componentWithId.container.querySelector('#test-id-tooltip-text')).toBeTruthy();
    expect(componentWithId.container).toMatchSnapshot();

  })

  //   expect(componentWithoutId.find('div').prop('aria-labelledby')).toBe(undefined);
  //   expect(componentWithId.find('div').prop('aria-labelledby')).toBe('test-id-tooltip-text');

  //   expect(componentWithId.find('span').prop('id')).toBe('test-id-tooltip-text');
  //   expect(componentWithoutId.find('span').prop('id')).toBe(undefined);

  //   componentWithId.unmount();
  //   componentWithoutId.unmount();
  // });
});
