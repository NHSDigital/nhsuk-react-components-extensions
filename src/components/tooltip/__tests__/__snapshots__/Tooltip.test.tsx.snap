// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Matches Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="nhsuk-tooltip"
      >
        Hover Over Me
        <span
          class="nhsuk-tooltip__text"
        >
          Tooltip!
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="nhsuk-tooltip"
    >
      Hover Over Me
      <span
        class="nhsuk-tooltip__text"
      >
        Tooltip!
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tooltip Passes through IDs 1`] = `
<div>
  <div
    class="nhsuk-tooltip"
  >
    Hover Over Me
    <span
      class="nhsuk-tooltip__text"
    >
      Tooltip!
    </span>
  </div>
</div>
`;

exports[`Tooltip Passes through IDs 2`] = `
<div>
  <div
    aria-labelledby="test-id-tooltip-text"
    class="nhsuk-tooltip"
    id="test-id"
  >
    Hover Over Me
    <span
      class="nhsuk-tooltip__text"
      id="test-id-tooltip-text"
    >
      Tooltip!
    </span>
  </div>
</div>
`;
